applications/MappingApplication/custom_mappers/nearest_neighbor_mapper_iga.h

```diff
@@ -0,0 +1,308 @@
+        : MapperInterfaceInfo(rCoordinates, SourceLocalSystemIndex, SourceRank),  
+        mShapeFunctionValues(),
+        mNearestNeighborId(),
+        mNumberOfNearestNeighbors(0),
+        mNearestNeighborDistance(std::numeric_limits<double>::max()) {}
+
+    MapperInterfaceInfo::Pointer Create() const override
+    {
+        return Kratos::make_shared<NearestNeighborInterfaceInfoIGA>();
+    }
+
+    MapperInterfaceInfo::Pointer Create(const CoordinatesArrayType& rCoordinates,
+                                        const IndexType SourceLocalSystemIndex,
+                                        const IndexType SourceRank) const override
+    {
+        return Kratos::make_shared<NearestNeighborInterfaceInfoIGA>(
+            rCoordinates,
+            SourceLocalSystemIndex,
+            SourceRank);
+    }
+
+    InterfaceObject::ConstructionType GetInterfaceObjectType() const override
+    {
+        return InterfaceObject::ConstructionType::Geometry_Center;
+    }
+
+    void ProcessSearchResult(const InterfaceObject& rInterfaceObject) override;
+
+    // Shape function values for the local system evaluated in the coordinates of the gauss point
+    void GetValue(std::vector<double>& rValue,
+                  const InfoType ValueType) const override
+    {
+        rValue = mShapeFunctionValues;
+    }
+    
+    // Nearest neighbor ids (control point ids) for the local system
+    void GetValue(std::vector<int>& rValue,
+                  const InfoType ValueType) const override
+    {
+        rValue = mNearestNeighborId;
+    }
+
+    void GetValue(double& rValue,
+                  const InfoType ValueType) const override
+    {
+        rValue = mNearestNeighborDistance;
+    }
+
+    void GetValue(IndexType& rValue,
+                  const InfoType ValueType) const override
+    {
+        rValue = mNumberOfNearestNeighbors;
+    }
+
+private:
+    std::vector<double> mShapeFunctionValues;
+    std::vector<int> mNearestNeighborId;
+    IndexType mNumberOfNearestNeighbors;
+    double mNearestNeighborDistance;
+
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS( rSerializer, MapperInterfaceInfo );
+        rSerializer.save("NearestNeighborId", mNearestNeighborId);
+        rSerializer.save("NearestNeighborDistance", mNearestNeighborDistance);
+        rSerializer.save("ShapeFunctionValues", mShapeFunctionValues);
+        rSerializer.save("NumberofNearestNeighbors", mNumberOfNearestNeighbors);
+    }
+
+    void load(Serializer& rSerializer) override
+    {
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS( rSerializer, MapperInterfaceInfo );
+        rSerializer.load("NearestNeighborId", mNearestNeighborId);
+        rSerializer.load("NearestNeighborDistance", mNearestNeighborDistance);
+        rSerializer.save("ShapeFunctionValues", mShapeFunctionValues);
+        rSerializer.save("NumberofNearestNeighbors", mNumberOfNearestNeighbors);
```

Comment:
beware of blind copy-pasta

```suggestion
        KRATOS_SERIALIZE_SAVE_BASE_CLASS( rSerializer, MapperInterfaceInfo );
        rSerializer.save("ShapeFunctionValues", mShapeFunctionValues);
        rSerializer.save("NearestNeighborId", mNearestNeighborId);
        rSerializer.save("NumberofNearestNeighbors", mNumberOfNearestNeighbors);
        rSerializer.save("NearestNeighborDistance", mNearestNeighborDistance);
    }

    void load(Serializer& rSerializer) override
    {
        KRATOS_SERIALIZE_LOAD_BASE_CLASS( rSerializer, MapperInterfaceInfo );
        rSerializer.load("ShapeFunctionValues", mShapeFunctionValues);
        rSerializer.load("NearestNeighborId", mNearestNeighborId);
        rSerializer.load("NumberofNearestNeighbors", mNumberOfNearestNeighbors);
        rSerializer.load("NearestNeighborDistance", mNearestNeighborDistance);
```